### PR TITLE
Keras model import update: embedding layer, end-to-end unit tests

### PR DESCRIPTION
**Pull request by turambar**
* Embedding layer
* end-to-end unit tests

### CODE REVIEW

File: deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        }
+        return predictions;
+    }
+
+    static public void compareINDArrays(String label, INDArray a, INDArray b, double eps) {
+        INDArray diff = a.sub(b);
+        double min = diff.minNumber().doubleValue();
+        double max = diff.maxNumber().doubleValue();
+        boolean eq = a.equalsWithEps(b, eps);
+//        log.info(label + ": " + a.equalsWithEps(b, eps) + ", " + min + ", " + max);
```
Comment (huitseeker) [inline deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java]: You may want to `log.debug` and leave this uncommented, no ?

File: deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for end-to-end Keras model import.
+ *
+ * TODO: find a more elegant fix for the per-layer activation comparisons
+ * since some layers (e.g., ActivationLayer) overwrite previous
+ * layer's activations when run in train=false mode.
+ *
+ * TODO: ndarray.equalsWithEps(ndarray) appears to be broken
```
Comment (huitseeker) [inline deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java]: In which way ? Is there an Nd4j issue ?

File: deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java (hunk 1 above)
Comment (turambar) [inline deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java]: yeah probably, these unit tests are WIP. ;)

File: deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java (hunk 2 above)
Comment (turambar) [inline deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java]: Not sure whether it's user error or a bug in ND4J. I just know that the stored Keras and computed DL4J activations are similar within 6-7 digits of precision, but `equalsWithEps` returns false. Investigating.

**Pull request merged**